Cannot build Tensile with ROCm's llvm fork on 3.5.0

I'm trying to build Tensile for rocBLAS using https://github.com/RadeonOpenCompute/llvm-project all on the `rocm-3.5.0` tag. I can't get it to build:

```
In file included from /tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/source/llvm/YAML.cpp:2:
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp:249:36: error: no viable conversion from 'llvm::StringRef' to 'const std::string' (aka 'const basic_string<char>')
                Impl::inputOne(io, key, *value);
                                   ^~~
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/Support/YAMLTraits.h:1101:31: note: in instantiation of member function 'llvm::yaml::CustomMappingTraits<llvm::yaml::Hide<std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>, std::less<std::__cxx11::basic_string<char>>, std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>>>>>>::inputOne' requested here
      CustomMappingTraits<T>::inputOne(io, key, Val);
                              ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp:96:13: note: in instantiation of function template specialization 'llvm::yaml::yamlize<llvm::yaml::Hide<std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>, std::less<std::__cxx11::basic_string<char>>, std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>>>>>>' requested here
            yamlize(io, hide, b, ctx);
            ^
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/Support/YAMLTraits.h:945:7: note: in instantiation of function template specialization 'llvm::yaml::yamlize<std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>, std::less<std::__cxx11::basic_string<char>>, std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>>>>>' requested here
      yamlize(*this, Val, Required, Ctx);
      ^
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/Support/YAMLTraits.h:851:11: note: in instantiation of function template specialization 'llvm::yaml::IO::processKey<std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>, std::less<std::__cxx11::basic_string<char>>, std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>>>>, llvm::yaml::EmptyContext>' requested here
    this->processKey(Key, Val, true, Ctx);
          ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp:131:20: note: in instantiation of function template specialization 'llvm::yaml::IO::mapRequired<std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>, std::less<std::__cxx11::basic_string<char>>, std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>>>>>' requested here
                io.mapRequired(key, obj);
                   ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/Serialization/ContractionSolution.hpp:62:22: note: (skipping 10 contexts in backtrace; use -ftemplate-backtrace-limit=0 to see all)
                iot::mapRequired(io, "info", s.info);
                     ^
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/Support/YAMLTraits.h:851:11: note: in instantiation of function template specialization 'llvm::yaml::IO::processKey<std::vector<std::shared_ptr<Tensile::ContractionSolution>, std::allocator<std::shared_ptr<Tensile::ContractionSolution>>>, llvm::yaml::EmptyContext>' requested here
    this->processKey(Key, Val, true, Ctx);
          ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp:131:20: note: in instantiation of function template specialization 'llvm::yaml::IO::mapRequired<std::vector<std::shared_ptr<Tensile::ContractionSolution>, std::allocator<std::shared_ptr<Tensile::ContractionSolution>>>>' requested here
                io.mapRequired(key, obj);
                   ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/Serialization/SolutionLibrary.hpp:148:22: note: in instantiation of function template specialization 'Tensile::Serialization::IOTraits<llvm::yaml::IO>::mapRequired<std::vector<std::shared_ptr<Tensile::ContractionSolution>, std::allocator<std::shared_ptr<Tensile::ContractionSolution>>>>' requested here
                iot::mapRequired(io, "solutions", solutions);
                     ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/Serialization/Base.hpp:209:46: note: in instantiation of member function 'Tensile::Serialization::MappingTraits<Tensile::MasterSolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>, llvm::yaml::IO, Tensile::Serialization::EmptyContext>::mapping' requested here
                MappingTraits<Subclass, IO>::mapping(io, *sc);
                                             ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp:265:52: note: in instantiation of function template specialization 'Tensile::Serialization::PointerMappingTraits<Tensile::MasterSolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>, llvm::yaml::IO, Tensile::Serialization::EmptyContext>::mapping<Tensile::MasterSolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>' requested here
                sn::PointerMappingTraits<obj, IO>::mapping(io, o);
                                                   ^
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/basic_string.h:448:7: note: candidate constructor not viable: no known conversion from 'llvm::StringRef' to 'const std::__cxx11::basic_string<char> &' for 1st argument
      basic_string(const basic_string& __str)
      ^
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/basic_string.h:525:7: note: candidate constructor not viable: no known conversion from 'llvm::StringRef' to 'const char *' for 1st argument
      basic_string(const _CharT* __s, const _Alloc& __a = _Alloc())
      ^
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/basic_string.h:552:7: note: candidate constructor not viable: no known conversion from 'llvm::StringRef' to 'std::__cxx11::basic_string<char> &&' for 1st argument
      basic_string(basic_string&& __str) noexcept
      ^
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/basic_string.h:579:7: note: candidate constructor not viable: no known conversion from 'llvm::StringRef' to 'initializer_list<char>' for 1st argument
      basic_string(initializer_list<_CharT> __l, const _Alloc& __a = _Alloc())
      ^
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/basic_string.h:440:7: note: explicit constructor is not a candidate
      basic_string(const _Alloc& __a) _GLIBCXX_NOEXCEPT
      ^
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/ADT/StringRef.h:275:14: note: explicit conversion function is not a candidate
    explicit operator std::string() const { return str(); }
             ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/Serialization/Containers.hpp:83:62: note: passing argument to parameter 'keyStr' here
            static void inputOne(IO & io, std::string const& keyStr, Map & value)
                                                             ^
In file included from /tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/source/llvm/Loading.cpp:29:
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp:249:36: error: no viable conversion from 'llvm::StringRef' to 'const std::string' (aka 'const basic_string<char>')
                Impl::inputOne(io, key, *value);
                                   ^~~
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/Support/YAMLTraits.h:1101:31: note: in instantiation of member function 'llvm::yaml::CustomMappingTraits<llvm::yaml::Hide<std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>, std::less<std::__cxx11::basic_string<char>>, std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>>>>>>::inputOne' requested here
      CustomMappingTraits<T>::inputOne(io, key, Val);
                              ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp:96:13: note: in instantiation of function template specialization 'llvm::yaml::yamlize<llvm::yaml::Hide<std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>, std::less<std::__cxx11::basic_string<char>>, std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>>>>>>' requested here
            yamlize(io, hide, b, ctx);
            ^
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/Support/YAMLTraits.h:945:7: note: in instantiation of function template specialization 'llvm::yaml::yamlize<std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>, std::less<std::__cxx11::basic_string<char>>, std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>>>>>' requested here
      yamlize(*this, Val, Required, Ctx);
      ^
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/Support/YAMLTraits.h:851:11: note: in instantiation of function template specialization 'llvm::yaml::IO::processKey<std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>, std::less<std::__cxx11::basic_string<char>>, std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>>>>, llvm::yaml::EmptyContext>' requested here
    this->processKey(Key, Val, true, Ctx);
          ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp:131:20: note: in instantiation of function template specialization 'llvm::yaml::IO::mapRequired<std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>, std::less<std::__cxx11::basic_string<char>>, std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char>>>>>' requested here
                io.mapRequired(key, obj);
                   ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/Serialization/ContractionSolution.hpp:62:22: note: (skipping 10 contexts in backtrace; use -ftemplate-backtrace-limit=0 to see all)
                iot::mapRequired(io, "info", s.info);
                     ^
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/Support/YAMLTraits.h:851:11: note: in instantiation of function template specialization 'llvm::yaml::IO::processKey<std::vector<std::shared_ptr<Tensile::ContractionSolution>, std::allocator<std::shared_ptr<Tensile::ContractionSolution>>>, llvm::yaml::EmptyContext>' requested here
    this->processKey(Key, Val, true, Ctx);
          ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp:131:20: note: in instantiation of function template specialization 'llvm::yaml::IO::mapRequired<std::vector<std::shared_ptr<Tensile::ContractionSolution>, std::allocator<std::shared_ptr<Tensile::ContractionSolution>>>>' requested here
                io.mapRequired(key, obj);
                   ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/Serialization/SolutionLibrary.hpp:148:22: note: in instantiation of function template specialization 'Tensile::Serialization::IOTraits<llvm::yaml::IO>::mapRequired<std::vector<std::shared_ptr<Tensile::ContractionSolution>, std::allocator<std::shared_ptr<Tensile::ContractionSolution>>>>' requested here
                iot::mapRequired(io, "solutions", solutions);
                     ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/Serialization/Base.hpp:209:46: note: in instantiation of member function 'Tensile::Serialization::MappingTraits<Tensile::MasterSolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>, llvm::yaml::IO, Tensile::Serialization::EmptyContext>::mapping' requested here
                MappingTraits<Subclass, IO>::mapping(io, *sc);
                                             ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp:265:52: note: in instantiation of function template specialization 'Tensile::Serialization::PointerMappingTraits<Tensile::MasterSolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>, llvm::yaml::IO, Tensile::Serialization::EmptyContext>::mapping<Tensile::MasterSolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>' requested here
                sn::PointerMappingTraits<obj, IO>::mapping(io, o);
                                                   ^
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/basic_string.h:448:7: note: candidate constructor not viable: no known conversion from 'llvm::StringRef' to 'const std::__cxx11::basic_string<char> &' for 1st argument
      basic_string(const basic_string& __str)
      ^
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/basic_string.h:525:7: note: candidate constructor not viable: no known conversion from 'llvm::StringRef' to 'const char *' for 1st argument
      basic_string(const _CharT* __s, const _Alloc& __a = _Alloc())
      ^
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/basic_string.h:552:7: note: candidate constructor not viable: no known conversion from 'llvm::StringRef' to 'std::__cxx11::basic_string<char> &&' for 1st argument
      basic_string(basic_string&& __str) noexcept
      ^
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/basic_string.h:579:7: note: candidate constructor not viable: no known conversion from 'llvm::StringRef' to 'initializer_list<char>' for 1st argument
      basic_string(initializer_list<_CharT> __l, const _Alloc& __a = _Alloc())
      ^
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/basic_string.h:440:7: note: explicit constructor is not a candidate
      basic_string(const _Alloc& __a) _GLIBCXX_NOEXCEPT
      ^
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/ADT/StringRef.h:275:14: note: explicit conversion function is not a candidate
    explicit operator std::string() const { return str(); }
             ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/Serialization/Containers.hpp:83:62: note: passing argument to parameter 'keyStr' here
            static void inputOne(IO & io, std::string const& keyStr, Map & value)
                                                             ^
In file included from /tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/source/llvm/YAML.cpp:2:
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp:249:36: error: no viable conversion from 'llvm::StringRef' to 'const std::string' (aka 'const basic_string<char>')
                Impl::inputOne(io, key, *value);
                                   ^~~
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/Support/YAMLTraits.h:1101:31: note: in instantiation of member function 'llvm::yaml::CustomMappingTraits<llvm::yaml::Hide<std::map<int, double, std::less<int>, std::allocator<std::pair<const int, double>>>>>::inputOne' requested here
      CustomMappingTraits<T>::inputOne(io, key, Val);
                              ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp:96:13: note: in instantiation of function template specialization 'llvm::yaml::yamlize<llvm::yaml::Hide<std::map<int, double, std::less<int>, std::allocator<std::pair<const int, double>>>>>' requested here
            yamlize(io, hide, b, ctx);
            ^
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/Support/YAMLTraits.h:945:7: note: in instantiation of function template specialization 'llvm::yaml::yamlize<std::map<int, double, std::less<int>, std::allocator<std::pair<const int, double>>>>' requested here
      yamlize(*this, Val, Required, Ctx);
      ^
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/Support/YAMLTraits.h:888:11: note: in instantiation of function template specialization 'llvm::yaml::IO::processKey<std::map<int, double, std::less<int>, std::allocator<std::pair<const int, double>>>, llvm::yaml::EmptyContext>' requested here
    this->processKey(Key, Val, false, Ctx);
          ^
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/Support/YAMLTraits.h:861:5: note: in instantiation of function template specialization 'llvm::yaml::IO::mapOptionalWithContext<std::map<int, double, std::less<int>, std::allocator<std::pair<const int, double>>>, llvm::yaml::EmptyContext>' requested here
    mapOptionalWithContext(Key, Val, Ctx);
    ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp:143:20: note: (skipping 11 contexts in backtrace; use -ftemplate-backtrace-limit=0 to see all)
                io.mapOptional(key, obj);
                   ^
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/Support/YAMLTraits.h:851:11: note: in instantiation of function template specialization 'llvm::yaml::IO::processKey<std::vector<std::shared_ptr<Tensile::ContractionSolution>, std::allocator<std::shared_ptr<Tensile::ContractionSolution>>>, llvm::yaml::EmptyContext>' requested here
    this->processKey(Key, Val, true, Ctx);
          ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp:131:20: note: in instantiation of function template specialization 'llvm::yaml::IO::mapRequired<std::vector<std::shared_ptr<Tensile::ContractionSolution>, std::allocator<std::shared_ptr<Tensile::ContractionSolution>>>>' requested here
                io.mapRequired(key, obj);
                   ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/Serialization/SolutionLibrary.hpp:148:22: note: in instantiation of function template specialization 'Tensile::Serialization::IOTraits<llvm::yaml::IO>::mapRequired<std::vector<std::shared_ptr<Tensile::ContractionSolution>, std::allocator<std::shared_ptr<Tensile::ContractionSolution>>>>' requested here
                iot::mapRequired(io, "solutions", solutions);
                     ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/Serialization/Base.hpp:209:46: note: in instantiation of member function 'Tensile::Serialization::MappingTraits<Tensile::MasterSolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>, llvm::yaml::IO, Tensile::Serialization::EmptyContext>::mapping' requested here
                MappingTraits<Subclass, IO>::mapping(io, *sc);
                                             ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp:265:52: note: in instantiation of function template specialization 'Tensile::Serialization::PointerMappingTraits<Tensile::MasterSolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>, llvm::yaml::IO, Tensile::Serialization::EmptyContext>::mapping<Tensile::MasterSolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>' requested here
                sn::PointerMappingTraits<obj, IO>::mapping(io, o);
                                                   ^
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/basic_string.h:448:7: note: candidate constructor not viable: no known conversion from 'llvm::StringRef' to 'const std::__cxx11::basic_string<char> &' for 1st argument
      basic_string(const basic_string& __str)
      ^
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/basic_string.h:525:7: note: candidate constructor not viable: no known conversion from 'llvm::StringRef' to 'const char *' for 1st argument
      basic_string(const _CharT* __s, const _Alloc& __a = _Alloc())
      ^
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/basic_string.h:552:7: note: candidate constructor not viable: no known conversion from 'llvm::StringRef' to 'std::__cxx11::basic_string<char> &&' for 1st argument
      basic_string(basic_string&& __str) noexcept
      ^
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/basic_string.h:579:7: note: candidate constructor not viable: no known conversion from 'llvm::StringRef' to 'initializer_list<char>' for 1st argument
      basic_string(initializer_list<_CharT> __l, const _Alloc& __a = _Alloc())
      ^
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/basic_string.h:440:7: note: explicit constructor is not a candidate
      basic_string(const _Alloc& __a) _GLIBCXX_NOEXCEPT
      ^
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/ADT/StringRef.h:275:14: note: explicit conversion function is not a candidate
    explicit operator std::string() const { return str(); }
             ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/Serialization/Containers.hpp:83:62: note: passing argument to parameter 'keyStr' here
            static void inputOne(IO & io, std::string const& keyStr, Map & value)
                                                             ^
In file included from /tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/source/llvm/Loading.cpp:29:
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp:249:36: error: no viable conversion from 'llvm::StringRef' to 'const std::string' (aka 'const basic_string<char>')
                Impl::inputOne(io, key, *value);
                                   ^~~
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/Support/YAMLTraits.h:1101:31: note: in instantiation of member function 'llvm::yaml::CustomMappingTraits<llvm::yaml::Hide<std::map<int, double, std::less<int>, std::allocator<std::pair<const int, double>>>>>::inputOne' requested here
      CustomMappingTraits<T>::inputOne(io, key, Val);
                              ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp:96:13: note: in instantiation of function template specialization 'llvm::yaml::yamlize<llvm::yaml::Hide<std::map<int, double, std::less<int>, std::allocator<std::pair<const int, double>>>>>' requested here
            yamlize(io, hide, b, ctx);
            ^
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/Support/YAMLTraits.h:945:7: note: in instantiation of function template specialization 'llvm::yaml::yamlize<std::map<int, double, std::less<int>, std::allocator<std::pair<const int, double>>>>' requested here
      yamlize(*this, Val, Required, Ctx);
      ^
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/Support/YAMLTraits.h:888:11: note: in instantiation of function template specialization 'llvm::yaml::IO::processKey<std::map<int, double, std::less<int>, std::allocator<std::pair<const int, double>>>, llvm::yaml::EmptyContext>' requested here
    this->processKey(Key, Val, false, Ctx);
          ^
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/Support/YAMLTraits.h:861:5: note: in instantiation of function template specialization 'llvm::yaml::IO::mapOptionalWithContext<std::map<int, double, std::less<int>, std::allocator<std::pair<const int, double>>>, llvm::yaml::EmptyContext>' requested here
    mapOptionalWithContext(Key, Val, Ctx);
    ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp:143:20: note: (skipping 11 contexts in backtrace; use -ftemplate-backtrace-limit=0 to see all)
                io.mapOptional(key, obj);
                   ^
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/Support/YAMLTraits.h:851:11: note: in instantiation of function template specialization 'llvm::yaml::IO::processKey<std::vector<std::shared_ptr<Tensile::ContractionSolution>, std::allocator<std::shared_ptr<Tensile::ContractionSolution>>>, llvm::yaml::EmptyContext>' requested here
    this->processKey(Key, Val, true, Ctx);
          ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp:131:20: note: in instantiation of function template specialization 'llvm::yaml::IO::mapRequired<std::vector<std::shared_ptr<Tensile::ContractionSolution>, std::allocator<std::shared_ptr<Tensile::ContractionSolution>>>>' requested here
                io.mapRequired(key, obj);
                   ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/Serialization/SolutionLibrary.hpp:148:22: note: in instantiation of function template specialization 'Tensile::Serialization::IOTraits<llvm::yaml::IO>::mapRequired<std::vector<std::shared_ptr<Tensile::ContractionSolution>, std::allocator<std::shared_ptr<Tensile::ContractionSolution>>>>' requested here
                iot::mapRequired(io, "solutions", solutions);
                     ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/Serialization/Base.hpp:209:46: note: in instantiation of member function 'Tensile::Serialization::MappingTraits<Tensile::MasterSolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>, llvm::yaml::IO, Tensile::Serialization::EmptyContext>::mapping' requested here
                MappingTraits<Subclass, IO>::mapping(io, *sc);
                                             ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp:265:52: note: in instantiation of function template specialization 'Tensile::Serialization::PointerMappingTraits<Tensile::MasterSolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>, llvm::yaml::IO, Tensile::Serialization::EmptyContext>::mapping<Tensile::MasterSolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>' requested here
                sn::PointerMappingTraits<obj, IO>::mapping(io, o);
                                                   ^
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/basic_string.h:448:7: note: candidate constructor not viable: no known conversion from 'llvm::StringRef' to 'const std::__cxx11::basic_string<char> &' for 1st argument
      basic_string(const basic_string& __str)
      ^
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/basic_string.h:525:7: note: candidate constructor not viable: no known conversion from 'llvm::StringRef' to 'const char *' for 1st argument
      basic_string(const _CharT* __s, const _Alloc& __a = _Alloc())
      ^
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/basic_string.h:552:7: note: candidate constructor not viable: no known conversion from 'llvm::StringRef' to 'std::__cxx11::basic_string<char> &&' for 1st argument
      basic_string(basic_string&& __str) noexcept
      ^
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/basic_string.h:579:7: note: candidate constructor not viable: no known conversion from 'llvm::StringRef' to 'initializer_list<char>' for 1st argument
      basic_string(initializer_list<_CharT> __l, const _Alloc& __a = _Alloc())
      ^
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/basic_string.h:440:7: note: explicit constructor is not a candidate
      basic_string(const _Alloc& __a) _GLIBCXX_NOEXCEPT
      ^
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/ADT/StringRef.h:275:14: note: explicit conversion function is not a candidate
    explicit operator std::string() const { return str(); }
             ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/Serialization/Containers.hpp:83:62: note: passing argument to parameter 'keyStr' here
            static void inputOne(IO & io, std::string const& keyStr, Map & value)
                                                             ^
In file included from /tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/source/llvm/YAML.cpp:2:
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp:249:36: error: no viable conversion from 'llvm::StringRef' to 'const std::string' (aka 'const basic_string<char>')
                Impl::inputOne(io, key, *value);
                                   ^~~
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/Support/YAMLTraits.h:1101:31: note: in instantiation of member function 'llvm::yaml::CustomMappingTraits<llvm::yaml::Hide<std::unordered_map<std::__cxx11::basic_string<char>, std::shared_ptr<Tensile::SolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>, std::hash<std::string>, std::equal_to<std::__cxx11::basic_string<char>>, std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::shared_ptr<Tensile::SolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>>>>>>::inputOne' requested here
      CustomMappingTraits<T>::inputOne(io, key, Val);
                              ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp:96:13: note: in instantiation of function template specialization 'llvm::yaml::yamlize<llvm::yaml::Hide<std::unordered_map<std::__cxx11::basic_string<char>, std::shared_ptr<Tensile::SolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>, std::hash<std::string>, std::equal_to<std::__cxx11::basic_string<char>>, std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::shared_ptr<Tensile::SolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>>>>>>' requested here
            yamlize(io, hide, b, ctx);
            ^
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/Support/YAMLTraits.h:945:7: note: in instantiation of function template specialization 'llvm::yaml::yamlize<std::unordered_map<std::__cxx11::basic_string<char>, std::shared_ptr<Tensile::SolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>, std::hash<std::string>, std::equal_to<std::__cxx11::basic_string<char>>, std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::shared_ptr<Tensile::SolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>>>>>' requested here
      yamlize(*this, Val, Required, Ctx);
      ^
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/Support/YAMLTraits.h:851:11: note: in instantiation of function template specialization 'llvm::yaml::IO::processKey<std::unordered_map<std::__cxx11::basic_string<char>, std::shared_ptr<Tensile::SolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>, std::hash<std::string>, std::equal_to<std::__cxx11::basic_string<char>>, std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::shared_ptr<Tensile::SolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>>>>, llvm::yaml::EmptyContext>' requested here
    this->processKey(Key, Val, true, Ctx);
          ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp:131:20: note: in instantiation of function template specialization 'llvm::yaml::IO::mapRequired<std::unordered_map<std::__cxx11::basic_string<char>, std::shared_ptr<Tensile::SolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>, std::hash<std::string>, std::equal_to<std::__cxx11::basic_string<char>>, std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::shared_ptr<Tensile::SolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>>>>>' requested here
                io.mapRequired(key, obj);
                   ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/Serialization/MapLibrary.hpp:48:22: note: (skipping 12 contexts in backtrace; use -ftemplate-backtrace-limit=0 to see all)
                iot::mapRequired(io, "map",      lib.map);
                     ^
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/Support/YAMLTraits.h:851:11: note: in instantiation of function template specialization 'llvm::yaml::IO::processKey<std::shared_ptr<Tensile::SolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>, llvm::yaml::EmptyContext>' requested here
    this->processKey(Key, Val, true, Ctx);
          ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp:131:20: note: in instantiation of function template specialization 'llvm::yaml::IO::mapRequired<std::shared_ptr<Tensile::SolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>>' requested here
                io.mapRequired(key, obj);
                   ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/Serialization/SolutionLibrary.hpp:173:22: note: in instantiation of function template specialization 'Tensile::Serialization::IOTraits<llvm::yaml::IO>::mapRequired<std::shared_ptr<Tensile::SolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>>' requested here
                iot::mapRequired(io, "library", innerLibrary);
                     ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/Serialization/Base.hpp:209:46: note: in instantiation of member function 'Tensile::Serialization::MappingTraits<Tensile::MasterSolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>, llvm::yaml::IO, Tensile::Serialization::EmptyContext>::mapping' requested here
                MappingTraits<Subclass, IO>::mapping(io, *sc);
                                             ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp:265:52: note: in instantiation of function template specialization 'Tensile::Serialization::PointerMappingTraits<Tensile::MasterSolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>, llvm::yaml::IO, Tensile::Serialization::EmptyContext>::mapping<Tensile::MasterSolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>' requested here
                sn::PointerMappingTraits<obj, IO>::mapping(io, o);
                                                   ^
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/basic_string.h:448:7: note: candidate constructor not viable: no known conversion from 'llvm::StringRef' to 'const std::__cxx11::basic_string<char> &' for 1st argument
      basic_string(const basic_string& __str)
      ^
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/basic_string.h:525:7: note: candidate constructor not viable: no known conversion from 'llvm::StringRef' to 'const char *' for 1st argument
      basic_string(const _CharT* __s, const _Alloc& __a = _Alloc())
      ^
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/basic_string.h:552:7: note: candidate constructor not viable: no known conversion from 'llvm::StringRef' to 'std::__cxx11::basic_string<char> &&' for 1st argument
      basic_string(basic_string&& __str) noexcept
      ^
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/basic_string.h:579:7: note: candidate constructor not viable: no known conversion from 'llvm::StringRef' to 'initializer_list<char>' for 1st argument
      basic_string(initializer_list<_CharT> __l, const _Alloc& __a = _Alloc())
      ^
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/basic_string.h:440:7: note: explicit constructor is not a candidate
      basic_string(const _Alloc& __a) _GLIBCXX_NOEXCEPT
      ^
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/ADT/StringRef.h:275:14: note: explicit conversion function is not a candidate
    explicit operator std::string() const { return str(); }
             ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/Serialization/Containers.hpp:83:62: note: passing argument to parameter 'keyStr' here
            static void inputOne(IO & io, std::string const& keyStr, Map & value)
                                                             ^
In file included from /tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/source/llvm/Loading.cpp:29:
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp:249:36: error: no viable conversion from 'llvm::StringRef' to 'const std::string' (aka 'const basic_string<char>')
                Impl::inputOne(io, key, *value);
                                   ^~~
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/Support/YAMLTraits.h:1101:31: note: in instantiation of member function 'llvm::yaml::CustomMappingTraits<llvm::yaml::Hide<std::unordered_map<std::__cxx11::basic_string<char>, std::shared_ptr<Tensile::SolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>, std::hash<std::string>, std::equal_to<std::__cxx11::basic_string<char>>, std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::shared_ptr<Tensile::SolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>>>>>>::inputOne' requested here
      CustomMappingTraits<T>::inputOne(io, key, Val);
                              ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp:96:13: note: in instantiation of function template specialization 'llvm::yaml::yamlize<llvm::yaml::Hide<std::unordered_map<std::__cxx11::basic_string<char>, std::shared_ptr<Tensile::SolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>, std::hash<std::string>, std::equal_to<std::__cxx11::basic_string<char>>, std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::shared_ptr<Tensile::SolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>>>>>>' requested here
            yamlize(io, hide, b, ctx);
            ^
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/Support/YAMLTraits.h:945:7: note: in instantiation of function template specialization 'llvm::yaml::yamlize<std::unordered_map<std::__cxx11::basic_string<char>, std::shared_ptr<Tensile::SolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>, std::hash<std::string>, std::equal_to<std::__cxx11::basic_string<char>>, std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::shared_ptr<Tensile::SolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>>>>>' requested here
      yamlize(*this, Val, Required, Ctx);
      ^
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/Support/YAMLTraits.h:851:11: note: in instantiation of function template specialization 'llvm::yaml::IO::processKey<std::unordered_map<std::__cxx11::basic_string<char>, std::shared_ptr<Tensile::SolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>, std::hash<std::string>, std::equal_to<std::__cxx11::basic_string<char>>, std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::shared_ptr<Tensile::SolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>>>>, llvm::yaml::EmptyContext>' requested here
    this->processKey(Key, Val, true, Ctx);
          ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp:131:20: note: in instantiation of function template specialization 'llvm::yaml::IO::mapRequired<std::unordered_map<std::__cxx11::basic_string<char>, std::shared_ptr<Tensile::SolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>, std::hash<std::string>, std::equal_to<std::__cxx11::basic_string<char>>, std::allocator<std::pair<const std::__cxx11::basic_string<char>, std::shared_ptr<Tensile::SolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>>>>>' requested here
                io.mapRequired(key, obj);
                   ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/Serialization/MapLibrary.hpp:48:22: note: (skipping 12 contexts in backtrace; use -ftemplate-backtrace-limit=0 to see all)
                iot::mapRequired(io, "map",      lib.map);
                     ^
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/Support/YAMLTraits.h:851:11: note: in instantiation of function template specialization 'llvm::yaml::IO::processKey<std::shared_ptr<Tensile::SolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>, llvm::yaml::EmptyContext>' requested here
    this->processKey(Key, Val, true, Ctx);
          ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp:131:20: note: in instantiation of function template specialization 'llvm::yaml::IO::mapRequired<std::shared_ptr<Tensile::SolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>>' requested here
                io.mapRequired(key, obj);
                   ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/Serialization/SolutionLibrary.hpp:173:22: note: in instantiation of function template specialization 'Tensile::Serialization::IOTraits<llvm::yaml::IO>::mapRequired<std::shared_ptr<Tensile::SolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>>' requested here
                iot::mapRequired(io, "library", innerLibrary);
                     ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/Serialization/Base.hpp:209:46: note: in instantiation of member function 'Tensile::Serialization::MappingTraits<Tensile::MasterSolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>, llvm::yaml::IO, Tensile::Serialization::EmptyContext>::mapping' requested here
                MappingTraits<Subclass, IO>::mapping(io, *sc);
                                             ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp:265:52: note: in instantiation of function template specialization 'Tensile::Serialization::PointerMappingTraits<Tensile::MasterSolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>, llvm::yaml::IO, Tensile::Serialization::EmptyContext>::mapping<Tensile::MasterSolutionLibrary<Tensile::ContractionProblem, Tensile::ContractionSolution>>' requested here
                sn::PointerMappingTraits<obj, IO>::mapping(io, o);
                                                   ^
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/basic_string.h:448:7: note: candidate constructor not viable: no known conversion from 'llvm::StringRef' to 'const std::__cxx11::basic_string<char> &' for 1st argument
      basic_string(const basic_string& __str)
      ^
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/basic_string.h:525:7: note: candidate constructor not viable: no known conversion from 'llvm::StringRef' to 'const char *' for 1st argument
      basic_string(const _CharT* __s, const _Alloc& __a = _Alloc())
      ^
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/basic_string.h:552:7: note: candidate constructor not viable: no known conversion from 'llvm::StringRef' to 'std::__cxx11::basic_string<char> &&' for 1st argument
      basic_string(basic_string&& __str) noexcept
      ^
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/basic_string.h:579:7: note: candidate constructor not viable: no known conversion from 'llvm::StringRef' to 'initializer_list<char>' for 1st argument
      basic_string(initializer_list<_CharT> __l, const _Alloc& __a = _Alloc())
      ^
/usr/lib/gcc/x86_64-linux-gnu/10/../../../../include/c++/10/bits/basic_string.h:440:7: note: explicit constructor is not a candidate
      basic_string(const _Alloc& __a) _GLIBCXX_NOEXCEPT
      ^
/home/harmen/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/llvm-amdgpu-3.5.0-2pca4hvik6g3rkshkawjtzhwmvv7xgvi/include/llvm/ADT/StringRef.h:275:14: note: explicit conversion function is not a candidate
    explicit operator std::string() const { return str(); }
             ^
/tmp/harmen/spack-stage/spack-stage-rocblas-3.5.0-mkl42uc35yef75uwoxnphdynuk2uzbfi/spack-build/virtualenv/lib/python3.7/site-packages/Tensile/Source/lib/include/Tensile/Serialization/Containers.hpp:83:62: note: passing argument to parameter 'keyStr' here
            static void inputOne(IO & io, std::string const& keyStr, Map & value)
                                                             ^

```

--- a/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp
+++ b/Tensile/Source/lib/include/Tensile/llvm/YAML.hpp
@@ -261,7 +261,7 @@ namespace llvm
 
             static void inputOne(IO& io, StringRef key, Hide<T>& value)
             {
-                Impl::inputOne(io, key, *value);
+                Impl::inputOne(io, key.str(), *value);
             }
 
             static void output(IO& io, Hide<T>& value)

